grype/presenter/config.go

```diff
@@ -0,0 +1,44 @@
+package presenter
+
+import "fmt"
+
+// Config is the presenter domain's configuration data structure.
+type Config struct {
+	format           format
+	templateFilePath string
+}
+
+// ValidatedConfig returns a new, validated presenter.Config. If a valid Config cannot be created using the given input,
+// an error is returned.
+func ValidatedConfig(output, outputTemplateFile string) (Config, error) {
+	format := parse(output)
+
+	if format == unknownFormat {
+		return Config{}, fmt.Errorf("unsupported output format %q, supported formats: %+v", output,
+			AvailableFormats)
+	}
+
+	if format == templateFormat {
+		if outputTemplateFile == "" {
+			return Config{}, fmt.Errorf("must specify path to template file when using %q output format",
+				templateFormat)
+		}
+
+		// TODO: Should we also validate access to the template file and the template's syntax, too?
```

Comment:
we can add this one day; I think it's acceptable to get this feature in without requiring up-front validation

Reply:
Okay, sounds good